### ISSUE TITLE
Libnetwork driver uses special `cali` prefix for interfaces

### ISSUE THREAD

**Issue by caseydavenport** [projectcalico/calico-containers#841]
Some users have tried to use libnetwork in conjunction with Kubernetes, which expects an 'eth0' interface and will fail if it isn't present.  Is there any reason we're not creating interfaces with the `eth` prefix?

From user mspreitz on Slack:

> BTW, I see that the Calico libnetwork plugin gives containers a network interface named `cali0`, not `eth0`.  FYI, I made a simple CNI plugin that connects to Docker networks (see https://github.com/kubernetes/kubernetes/pull/21956).  Kubernetes seems to have a problem when the container's network interface is not named `eth0`.  My kubelet is continually making complaints like this:

```
E0224 22:16:44.536571   18072 manager.go:377] NetworkPlugin cni failed on the status hook for pod 'kubernetes-dashboard-v0.1.0-zjjq7' - invalid CIDR address: Device "eth0" does not exist.
I0224 22:16:44.539020   18072 manager.go:1769] pod "kubernetes-dashboard-v0.1.0-zjjq7_kube-system" container "kubernetes-dashboard" is unhealthy (probe result: failure), it will be killed and re-created.
```

> Manually doing the health test of that pod gets a successful result, so I suppose the "unhealthy" determination is a consequence of the network interface name confusion.

**Comment by caseydavenport** [comment]
@tomdee @MikeSpreitzer looks like I opened this a while ago!

Let's fix this up in time for the next release.

**Comment by MikeSpreitzer** [comment]
Thanks.

Just to be utterly clear, we are talking about the name of the network interface inside the container, not its compatriot outside the container.

**Comment by caseydavenport** [comment]
Fixed by https://github.com/projectcalico/libnetwork-plugin/pull/88

**Comment by MikeSpreitzer** [comment]
I use `calicoctl node --libnetwork`.  So can I get the new functionality by changing that to `CALICO_LIBNETWORK_IFPREFIX=eth calicoctl node --libnetwork` ?

**Comment by caseydavenport** [comment]
@gunjan5 can you ensure that in v1.0.0 we're passing the CALICO_LIBNETWORK_IFPREFIX value through via calicoctl?

**Comment by gunjan5** [comment]
fixed in #1390
